scylla-cdc-replicator/src/replicator_consumer.rs

```diff
@@ -0,0 +1,248 @@
+        let names = keys_iter.clone().join(",");
+        let markers = keys_iter.clone().map(|_| "?").join(",");
+
+        let insert_query = dest_session
+            .prepare(format!(
+                "INSERT INTO {}.{} ({}) VALUES ({}) USING TTL ?",
+                dest_keyspace_name, dest_table_name, names, markers
+            ))
+            .await
+            .expect("Preparing insert query failed.");
+
+        let keys_cond = keys_iter.map(|name| format!("{} = ?", name)).join(" AND ");
+
+        ReplicatorConsumer {
+            dest_session,
+            dest_keyspace_name,
+            dest_table_name,
+            table_schema,
+            non_key_columns,
+            insert_query,
+            keys_cond,
+        }
+    }
+
+    fn get_keys_iter(table_schema: &Table) -> impl std::iter::Iterator<Item = &String> + Clone {
+        table_schema
+            .partition_key
+            .iter()
+            .chain(table_schema.clustering_key.iter())
+    }
+
+    fn get_timestamp(data: &CDCRow<'_>) -> i64 {
+        const NANOS_IN_MILLIS: u64 = 1000;
+        (data.time.to_timestamp().unwrap().to_unix_nanos() / NANOS_IN_MILLIS) as i64
+    }
+
+    async fn update(&self, data: CDCRow<'_>) {
+        self.update_or_insert(data, false).await;
+    }
+
+    async fn insert(&self, data: CDCRow<'_>) {
+        self.update_or_insert(data, true).await;
+    }
+
+    async fn update_atomic_or_frozen<'a>(
+        &self,
+        column_name: &str,
+        data: &'a CDCRow<'_>,
+        mut values_for_update: Vec<&'a CqlValue>,
+        values_for_delete: &[&CqlValue],
+        timestamp: i64,
+    ) {
+        if let Some(value) = data.get_value(column_name) {
+            // Order of values: ttl, inserted value, pk condition values.
+            values_for_update[1] = value;
+            self.run_statement(
+                Query::new(format!(
+                    "UPDATE {}.{} USING TTL ? SET {} = ? WHERE {}",
+                    self.dest_keyspace_name, self.dest_table_name, column_name, self.keys_cond
+                )),
+                &values_for_update,
+                timestamp,
+            )
+            .await;
+        } else if data.is_value_deleted(column_name) {
+            self.run_statement(
+                Query::new(format!(
+                    "DELETE {} FROM {}.{} WHERE {}",
+                    column_name, self.dest_keyspace_name, self.dest_table_name, self.keys_cond
+                )),
+                values_for_delete,
+                timestamp,
+            )
+            .await;
+        }
+    }
+
+    async fn update_or_insert(&self, data: CDCRow<'_>, is_insert: bool) {
+        let keys_iter = ReplicatorConsumer::get_keys_iter(&self.table_schema);
+        let ttl = CqlValue::Int(data.ttl.unwrap_or(0) as i32); // If data is inserted without TTL, setting it to 0 deletes existing TTL.
+        let timestamp = ReplicatorConsumer::get_timestamp(&data);
```

Comment:
There was a problem with TTL. When I used `CqlValue:Bigint`, I received the following error:
```
DbError(Invalid, "Exception while binding column \"[ttl]\": marshaling error: Validation failed for type org.apache.cassandra.db.marshal.Int32Type: got 8 bytes")
```
This is strange, because CDC table holds ttl as bigint (which consists of 8 bytes).